kolibri/core/auth/management/commands/bulkimportusers.py

```diff
@@ -0,0 +1,587 @@
+import csv
+import logging
+import re
+import sys
+
+from django.contrib.auth.hashers import make_password
+from django.core.exceptions import ValidationError
+from django.core.management.base import CommandError
+
+from kolibri.core.auth.constants.demographics import choices
+from kolibri.core.auth.csv_utils import input_fields
+from kolibri.core.auth.csv_utils import labels
+from kolibri.core.auth.models import Classroom
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import FacilityUser
+from kolibri.core.auth.models import Membership
+from kolibri.core.tasks.management.commands.base import AsyncCommand
+from kolibri.core.tasks.utils import get_current_job
+
+logger = logging.getLogger(__name__)
+DEFAULT_PASSWORD = make_password("kolibri")
+fieldnames = (
+    "Username",
+    "Password",
+    "Full name",
+    "User type",
+    "Identifier",
+    "Birth year",
+    "Gender",
+    "Enrolled in",
+    "Assigned to",
+)
+roles_map = {
+    "learner": None,
+    "admin": "admin",
+    "facility coach": "coach",
+    "class coach": "classroom assignable coach",
+}
+
+# Validators ###
+
+
+def number_range(min, max):
+    """
+    Return a value check function which raises a ValueError if the supplied
+    value is less than `min` or greater than `max`.
+    """
+
+    def checker(v):
+        if int(v) < min or int(v) > max:
+            raise ValueError(v)
+
+    return checker
+
+
+def value_length(length, null=False):
+    """
+    Return a value check function which raises a ValueError if the supplied
+    value has a length greater than 'length'
+    If null is not True raises a ValueError if the supplied value is None.
+    """
+
+    def checker(v):
+        if null and v is None:
+            return checker
```

Comment:
I'm a bit confused what's happening here. Shouldn't this just be `return`, not `return checker`?

Reply:
no, it should return `checker` as validators are functions that are passed later, see function `csv_values_validation` on line 244. It works similarly to decorators. Examples:

```
        validator.add_check("Birth year", number_range(1900, 99999), "Not a valid year")
        validator.add_check("Username", value_length(125), "User name is too long")
```